server/fleet/scripts.go

```diff
@@ -29,6 +29,90 @@ func (s Script) AuthzType() string {
 	return "script"
 }
 
+// HostScripDetail represents the details of a script that applies to a specific host.
+type HostScriptDetail struct {
+	// HostID is the ID of the host.
+	HostID uint `json:"-"`
+	// ScriptID is the ID of the script.
+	ScriptID uint `json:"script_id"`
+	// Name is the name of the script.
+	Name string `json:"name"`
+	// LastExecution is the most recent execution of the script on the host. It is nil if the script
+	// has never executed on the host.
+	LastExecution *HostScriptExecution `json:"last_execution"`
+}
+
+// NewHostScriptDetail creates a new HostScriptDetail and sets its LastExecution field based on the
+// provided details.
+func NewHostScriptDetail(hostID, scriptID uint, name string, executionID *string, executedAt *time.Time, exitCode *int64, hsrID *uint) *HostScriptDetail {
+	hs := HostScriptDetail{
+		HostID:   hostID,
+		ScriptID: scriptID,
+		Name:     name,
+	}
+	hs.SetLastExecution(executionID, executedAt, exitCode, hsrID)
+	return &hs
+}
+
+// HostScriptExecution represents a single execution of a script on a host.
+type HostScriptExecution struct {
+	// HostID is the ID of the host.
+	HostID uint `json:"-"`
+	// ScriptID is the ID of the script.
+	ScriptID uint `json:"-"`
+	// HSRID is the unique row identifier of the host_script_results table for this execution.
+	HSRID uint `json:"-"`
+	// ExecutionID is a unique identifier for a single execution of the script.
+	ExecutionID string `json:"execution_id"`
+	// ExecutedAt represents the time that the script was executed on the host. It should correspond to
+	// the created_at field of the host_script_results table for the associated HSRID.
+	ExecutedAt time.Time `json:"executed_at"`
+	// Status is the status of the script execution. It is one of "pending", "ran", or "error". It
+	// is derived from the exit_code field of the host_script_results table for the associated HSRID.
+	Status string `json:"status"`
+}
+
+// SetLastExecution updates the LastExecution field of the HostScriptDetail if the provided details
+// are more recent than the current LastExecution. It returns true if the LastExecution was updated.
+func (hs *HostScriptDetail) SetLastExecution(executionID *string, executedAt *time.Time, exitCode *int64, hsrID *uint) bool {
```

Comment:
I _think_ this is only used internally in this package? If so I'd make it unexported. (I may have missed some uses in tests though)